### ISSUE TITLE
add_phylopic_base Fails with Complex layout()

### ISSUE THREAD

**Issue by willgearty** [sckott/rphylopic#14]
Hey Scott,
I've been working on some figures for a mansucript, and up to now everything had been going smoothly.
I was able to insert some phylopics into my individual figures with ease using your package.
However, when I started combining figures into complex layouts, the `add_phylopic_base` function seemed to have some trouble.
I ran your example with no problems:

```
posx <- runif(50, 0, 10)
posy <- runif(50, 0, 10)
sizey <- runif(50, 0.01, 0.5)
cols <- sample(c("black", "darkorange", "grey42", "white"), 50, replace = TRUE)

cat <- image_data("23cd6aa4-9587-4a2e-8e26-de42885004c9", size = 128)[[1]]
plot(posx, posy, type="n", main="A cat herd")
for (i in 1:50) {
  add_phylopic_base(cat, 1, posx[i], posy[i], sizey[i], cols[i])
}
```

However, if you try that in a more complex layout, it doesn't seem to plot anything (that I can see):

```
layout(matrix(c(1,2,3,4), nrow=2,ncol=2,byrow=TRUE))
posx <- runif(50, 0, 10)
posy <- runif(50, 0, 10)
sizey <- runif(50, 0.01, 0.5)
cols <- sample(c("black", "darkorange", "grey42", "white"), 50, replace = TRUE)

cat <- image_data("23cd6aa4-9587-4a2e-8e26-de42885004c9", size = 128)[[1]]
for(j in 1:4){
    plot(posx, posy, type="n", main="A cat herd")
    for (i in 1:50) {
      add_phylopic_base(cat, 1, posx[i], posy[i], sizey[i], cols[i])
    }
}
```

Other attempts produced the images, but in incorrect locations (and often smaller than requested).

I've had luck using the code here: https://scrogster.wordpress.com/2014/06/02/adding-phylopic-org-silhouettes-to-r-plots/.
You referenced that link in a different issue, and the code there seems to work fine with complex layouts.

Just figured I would let you know,
Will

**Comment by sckott** [comment]
hi @willgearty - thanks for this.  I'll have a look at that blog post and see how we can make the base plots stuff work better!

**Comment by sckott** [comment]
@willgearty reinstall and try again, used the code from that post

@dill didn't take time to see why your original code doesn't work in some cases, but can try later

**Comment by dill** [comment]
@sckott my original solution was goofy. I looked up how to do something better, pulled the new code and then saw you'd done exactly what I was going to.

Sorry for the inconvenience @willgearty!

**Comment by willgearty** [comment]
No worries, I ended up having to write my own code based on that code anyways, because there were some sizing problems:
```
logoing_func<-function(logo, x, y, size){
  dims<-dim(logo)[1:2] #number of x-y pixels for the logo
  IAR<-dims[1]/dims[2] #image aspect ratio
  limits <- par("usr")
  AAR<-abs(limits[3]-limits[4])/abs(limits[1]-limits[2]) #axes aspect ratio
  PAR<-par("pin")[2]/par("pin")[1]
  polygon(c(0,0),c(0,0), col=rgb(0,0,0,alpha = 1))
  rasterImage(logo, x-(size/2), y-(IAR*AAR/PAR*size/2), x+(size/2), y+(IAR*AAR/PAR*size/2), interpolate = FALSE)
}
```
This allows for the user to specify exactly where they want the image using the plot's x and y dimensions.
It then rescales the image based on the axes aspect ratio and the plot area aspect ratio.
Figured I would share in case anyone else runs into a similar situation.

Another odd occurrence using the blog post code was that the image would inherit the transparency of whatever had been plotted before it.
In my case, I had been plotting some polygons with `alpha = .5`, which then caused the image to have .5 transparency as well.
Plotting an empty polygon with `alpha = 1` before `rasterImage()` fixed this odd problem for me (see code above).
I'm not sure what part of `rasterImage()` is inheriting the previous transparency, and I'm not sure where to report that problem (or if it even is a problem).
